src/context/auth-context.js

```diff
@@ -0,0 +1,87 @@
+import { createContext, useContext, useState } from "react";
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+
+const AuthContext = createContext(null);
+
+const AuthProvider = ({ children }) => {
+  const localStorageToken = JSON.parse(localStorage.getItem("token"));
+  const [token, setToken] = useState(localStorageToken?.token);
+  const localStorageUser = JSON.parse(localStorage.getItem("user"));
+  const [user, setUser] = useState(localStorageUser?.user);
+  const [tokenState, setTokenState] = useState(false);
+  const navigate = useNavigate();
+
+  const signUpUser = async (email, password, firstname, lastname) => {
+    try {
+      const {
+        data: { createdUser, encodedToken },
+        status,
+      } = await axios.post("/api/auth/signup", {
+        email: email,
+        password: password,
+        firstName: firstname,
+        lastName: lastname,
+      });
+      if (status === 200 || 201) {
+        localStorage.setItem("signup", JSON.stringify({ token: encodedToken }));
+        setToken(encodedToken);
+        localStorage.setItem("user", JSON.stringify({ user: createdUser }));
+        setUser(createdUser);
+        alert("Successfully Created! Proceed to login with your credentials");
+        setTimeout(() => {
```

Comment:
don't think there is a need to naviagete using setTimeout. Remove the setTimeout as its already async the function